modules/lib/os.nix

```diff
@@ -0,0 +1,15 @@
+{
+  isNixOS =
```

Comment:
Unfortunately this will not work since a HM configuration may be built on one host and deployed on another. I would instead suggest adding an option along the lines of
```
home.targetOperatingSystem = mkOption {
  type = types.enum [ "nixos" "linux" ];
  default = "nixos";
  example = "linux";
  description = ''
    The operating system on which this configuration will be deployed.
  '';
};
```

The installation script could look in `/etc/os-release` and include
```
home.targetOperatingSystem = "linux";
```
in the generated initial configuration if `/etc/os-release` has `ID != "nixos"`.